### PR TITLE
Fix duplicate CI runs for both push and pull_request events

### PR DESCRIPTION
**Pull request by vatsalghelani-csa**
Disable CI runs for push event on any branch except `master` and `v*-branch`. This means if we want CI runs on branches not from a fork of connectedhomeip, we will need to make pull requests.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -16,6 +16,9 @@ name: Builds
 
 on:
     push:
+        branches:
+            - master
+            - 'v*-branch'
         branches-ignore:
```
Comment (andy31415) [inline .github/workflows/build.yaml:22]: do we need an ignore at all given that we have `branches` element?

Comment (vatsalghelani-csa) [reply .github/workflows/build.yaml]: Removed the `branches-ignore` element from the push event.

**Review by andy31415 — APPROVED** [review]

**Review by arkq — APPROVED** [review]

**Comment by andy31415** [comment]
@vatsalghelani-csa the previous run said "EFR32" did not run. I updated to master to kick CI to see if this is a real issue or not...

File: .github/workflows/examples-efr32.yaml [hunk 2]
```diff
@@ -15,9 +15,10 @@
 name: Build example - EFR32
 
 on:
-  push:
-    branches-ignore:
-      - 'dependabot/**'
+    push:
```
Comment (arkq) [inline .github/workflows/examples-efr32.yaml:18]: Maybe the problem is with inconsistent indentation? The rest of this file is indented with 2 spaces, not 4.

Comment (vatsalghelani-csa) [reply .github/workflows/examples-efr32.yaml:18]: @arkq I would agree with you on this. I noticed that too, but given the file’s long-standing state, I hesitated to make the indentation changes with this commit.

Comment (arkq) [reply .github/workflows/examples-efr32.yaml:18]: Yes, but YAML should be indented by 2 spaces (at least that's what most YAML linters do)

File: .github/workflows/examples-efr32.yaml [hunk 3]
```diff
@@ -15,10 +15,9 @@
 name: Build example - EFR32
 
 on:
-    push:
-        branches:
-            - master
-            - 'v*-branch'
+  push:
+    branches-ignore:
```
Comment (arkq) [inline .github/workflows/examples-efr32.yaml:19]: I think that you can make changes here as long as you will not change the indentation (keep it as it is right now - 2 spaces).

**Pull request merged**